metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/dao/ElasticsearchDao.java

```diff
@@ -102,15 +128,99 @@ public SearchResponse search(SearchRequest searchRequest) throws InvalidSearchEx
       return searchResult;
     }).collect(Collectors.toList()));
     return searchResponse;
   }
 
   @Override
-  public void init(Map<String, Object> globalConfig, AccessConfig config) {
-    this.client = ElasticsearchUtils.getClient(globalConfig, config.getOptionalSettings());
-    this.accessConfig = config;
+  public synchronized void init(AccessConfig config) {
+    if(this.client == null) {
+      this.client = ElasticsearchUtils.getClient(config.getGlobalConfigSupplier().get(), config.getOptionalSettings());
+      this.accessConfig = config;
+    }
+  }
+
+  @Override
+  public Document getLatest(final String guid, final String sensorType) throws IOException {
+    Optional<Document> ret = searchByGuid(
+            guid
+            , sensorType
+            , hit -> {
+              Long ts = 0L;
+              String doc = hit.getSourceAsString();
+              String sourceType = Iterables.getFirst(Splitter.on("_doc").split(hit.getType()), null);
+              try {
+                return Optional.of(new Document(doc, guid, sourceType, ts));
+              } catch (IOException e) {
+                throw new IllegalStateException("Unable to retrieve latest: " + e.getMessage(), e);
+              }
+            }
+            );
+    return ret.orElse(null);
+  }
+
+  /**
+   * Return the search hit based on the UUID and sensor type.
+   * A callback can be specified to transform the hit into a type T.
+   * If more than one hit happens, the first one will be returned.
+   * @throws IOException
+   */
+  <T> Optional<T> searchByGuid(String guid, String sensorType, Function<SearchHit, Optional<T>> callback) throws IOException{
+    QueryBuilder query =  QueryBuilders.matchQuery(Constants.GUID, guid);
+    SearchRequestBuilder request = client.prepareSearch()
+                                         .setTypes(sensorType + "_doc")
+                                         .setQuery(query)
+                                         .setSource("message")
+                                         ;
+    MultiSearchResponse response = client.prepareMultiSearch()
+                                         .add(request)
+                                         .get();
+    for(MultiSearchResponse.Item i : response) {
+      org.elasticsearch.action.search.SearchResponse resp = i.getResponse();
+      SearchHits hits = resp.getHits();
+      for(SearchHit hit : hits) {
+        Optional<T> ret = callback.apply(hit);
+        if(ret.isPresent()) {
+          return ret;
+        }
+      }
+    }
+    return Optional.empty();
+
+  }
+
+  @Override
+  public void update(Document update, Optional<String> index) throws IOException {
+    String indexPostfix = ElasticsearchUtils.getIndexFormat(accessConfig.getGlobalConfigSupplier().get()).format(new Date());
+    String sensorType = update.getSensorType();
+    String indexName = ElasticsearchUtils.getIndexName(sensorType, indexPostfix, null);
+
+    String type = sensorType + "_doc";
```

Comment:
Thinking about it for a minute, I think we need to create a `README.md` in `metron-elasticsearch` that details some of the assumptions for how messages are written in ES.  I'll create a JIRA around that.